src/main/java/scanner/Scanner.java

```diff
@@ -277,12 +288,30 @@ boolean lexicalAnalysis(){
 			System.out.println(getTokenString(token));
 			// falls kein gültiges Token gefunden wurde, lexikalische Analyse
 			// abbrechen
-			if (token==NO_TYPE)
+			if (token==NO_TYPE) {
 				return false;
+			}
 			// sonst Token in tokenStream eintragen
-			else
-				tokenStream.
-				addLast(new Token(token,inputStream.get(pointer-1).line,lexem));
+			else {
+				System.out.println("LEXEM: " + lexem);
+				System.out.println("token: " + getTokenString(token));
+				if(token==IDENT) {
+					System.out.println("token: " + getTokenString(token));
+					if(lexem.equals(new String("is"))) {
+						token=VARIABLE_ASSIGMENT;
+						System.out.println("token: " + getTokenString(token));
+
+					}
```

Comment:
Wir erinnern uns an Methoden tun eine einzige Sache. Das ist ein Beispiel für einen Block der in eine eigene Funktionseinheit exportiert werden kann. Das ist nur ein Bsp. für alle anderen Möglichkeiten/ Notwendigkeiten das im Code zu wiederholen.